src/utils/lbann_library.cpp

```diff
@@ -387,4 +387,11 @@ void print_lbann_configuration(lbann_comm *comm, int io_threads_per_process, int
   std::cout << std::endl;
 }
 
+void lbann_mpi_err_handler(MPI_Comm *comm, int *err_code, ... ) {
+  char err_string[1024];
+  int err_string_length;
+  MPI_Error_string(*err_code, &err_string[0], &err_string_length);
+  LBANN_ERROR("MPI threw this error: ", err_string);
```

Comment:
Pro: This will give you a stack trace.
Con: The `__FILE__` and `__LINE__` expansion will always point to `lbann_library.cpp:394`.